packages/flutter_tools/test/general.shard/macos/darwin_dependency_management_test.dart

```diff
@@ -5,37 +5,52 @@
 import 'package:file/file.dart';
 import 'package:file/memory.dart';
 import 'package:flutter_tools/src/base/logger.dart';
+import 'package:flutter_tools/src/flutter_manifest.dart';
 import 'package:flutter_tools/src/macos/cocoapods.dart';
 import 'package:flutter_tools/src/macos/darwin_dependency_management.dart';
 import 'package:flutter_tools/src/macos/swift_package_manager.dart';
 import 'package:flutter_tools/src/platform_plugins.dart';
 import 'package:flutter_tools/src/plugins.dart';
 import 'package:flutter_tools/src/project.dart';
 import 'package:test/fake.dart';
+import 'package:unified_analytics/unified_analytics.dart';
 
 import '../../src/common.dart';
+import '../../src/fakes.dart';
 
 void main() {
   const List<SupportedPlatform> supportedPlatforms = <SupportedPlatform>[
     SupportedPlatform.ios,
     SupportedPlatform.macos,
   ];
 
+  late MemoryFileSystem testFileSystem;
+  late BufferLogger testLogger;
+  late FakeAnalytics fakeAnalytics;
+
+  setUp(() {
+    testFileSystem = MemoryFileSystem.test();
+    testLogger = BufferLogger.test();
+    fakeAnalytics = getInitializedFakeAnalyticsInstance(
+      fs: testFileSystem,
+      fakeFlutterVersion: FakeFlutterVersion(),
+    );
+  });
```

Comment:
Flutter's style guide recommends avoiding `setUp`

> Avoid using setUp, tearDown, and similar features, as well as test-global variables or other state shared between tests. They make writing tests easier but make maintaining them, debugging them, and refactoring code much harder.

https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md

Reply:
Thanks for the link - updated these to remove `setUp`!